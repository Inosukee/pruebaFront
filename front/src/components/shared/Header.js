import React, { useEffect } from 'react'
import queryString from 'query-string';
import logo from '../../assets/img/Logo_ML.png'
import searchIco from '../../assets/img/ic_Search.png'
import { useForm } from '../../hooks/useForm';
import { Link, useLocation } from 'react-router-dom';

export const Header = ({ setFilter, history }) => {
    const location = useLocation();
    const { search = '' } = queryString.parse(location.search);

    const [formValues, handleInputChange] = useForm({
        searchText: search
    });

    const { searchText } = formValues;

    useEffect(() => setFilter(search), [setFilter, search])

    const handleSubmit = (e) => {
        e.preventDefault();
        if (searchText.trim().length > 0) {
            if (location.pathname.includes('/items/')) {
                history.push(`/items?search=${searchText}`);
            } else {
                history.push(`items?search=${searchText}`);
            }
        }

    }
    return (
        <header role='banner' className='header__header'>
            <Link to={'/'} className='pointer'>
                <img className='header__img' src={logo} alt="logo" />
            </Link>
            <form onSubmit={handleSubmit} className="header__group">
                <input type="text" className="header__input" placeholder="Nunca dejes de buscar" value={searchText}
                    onChange={handleInputChange} name="searchText" />
                <button className="btn btn-outline-secondary header__button" type="submit"><img src={searchIco} alt="search" /></button>
            </form>
        </header>
    )
}
