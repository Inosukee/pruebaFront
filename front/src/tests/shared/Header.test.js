import React from 'react'
import { shallow } from 'enzyme';
import { Header } from '../../components/shared/Header';



jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "localhost:3000"
  })
}));


describe('Pruebas en <Header/>', () => {
  const wrapper = shallow(<Header />)

  test('debe mostrar el componente correctamente', () => {
    expect(wrapper.find('.header__input').exists()).toBe(true)
    expect(wrapper.find('.header__img').exists()).toBe(true)  
    expect(wrapper).toMatchSnapshot()
  })

  test('debe de cambiar el texto', () => {
    const input = wrapper.find('input')
    const value = 'testInput';
    const name = 'searchText';
    input.simulate('change', { target: { name:name, value:value } })

  })

})


