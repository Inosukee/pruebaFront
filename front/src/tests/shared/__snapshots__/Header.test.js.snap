// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Header/> debe mostrar el componente correctamente 1`] = `
<header
  className="header__header"
  role="banner"
>
  <Link
    className="pointer"
    to="/"
  >
    <img
      alt="logo"
      className="header__img"
      src="Logo_ML.png"
    />
  </Link>
  <form
    className="header__group"
    onSubmit={[Function]}
  >
    <input
      className="header__input"
      name="searchText"
      onChange={[Function]}
      placeholder="Nunca dejes de buscar"
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-secondary header__button"
      type="submit"
    >
      <img
        alt="search"
        src="ic_Search.png"
      />
    </button>
  </form>
</header>
`;
